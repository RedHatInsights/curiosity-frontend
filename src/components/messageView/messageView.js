import React from 'react';
import PropTypes from 'prop-types';
import {
  EmptyState as PfEmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import { PageLayout, PageHeader } from '../pageLayout/pageLayout';
import { helpers } from '../../common';

/**
 * FixMe: Patternfly EmptyStateIcon PropType registers as function?
 */
/**
 * Render a message view.
 *
 * @param {object} props
 * @param {Node|Function} props.icon
 * @param {string} props.message
 * @param {string} props.pageTitle
 * @param {string} props.title
 * @returns {Node}
 */
const MessageView = ({ icon, message, pageTitle, title }) => (
  <PageLayout>
    <PageHeader>{pageTitle || helpers.UI_DISPLAY_NAME}</PageHeader>
    <PfEmptyState variant={EmptyStateVariant.full} className="fadein">
      {title && (
        <Title headingLevel="h2" size="lg">
          {title}
        </Title>
      )}
      {icon && <EmptyStateIcon icon={icon} />}
      {message && <EmptyStateBody>{message}</EmptyStateBody>}
    </PfEmptyState>
  </PageLayout>
);

/**
 * Prop types.
 *
 * @type {{icon: Node|Function, message: string, title: string}}
 */
MessageView.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  message: PropTypes.string,
  pageTitle: PropTypes.string,
  title: PropTypes.string
};

/**
 * Default props.
 *
 * @type {{icon: null, message: null, title: null}}
 */
MessageView.defaultProps = {
  icon: null,
  message: null,
  pageTitle: null,
  title: null
};

export { MessageView as default, MessageView };
