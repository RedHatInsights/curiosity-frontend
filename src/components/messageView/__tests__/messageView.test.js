import React from 'react';
import { shallow } from 'enzyme';
import { BanIcon } from '@patternfly/react-icons';
import { MessageView } from '../messageView';

describe('MessageView Component', () => {
  it('should render a non-connected component', () => {
    const props = {
      icon: BanIcon,
      message: 'ipsum',
      title: 'lorem'
    };

    const component = shallow(<MessageView {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });

  it('should have fallback conditions for all props', () => {
    const props = {};

    const component = shallow(<MessageView {...props} />);
    expect(component).toMatchSnapshot('fallback display');
  });

  it('should render children when provided', () => {
    const props = { children: 'child content' };

    const component = shallow(<MessageView {...props} />);
    expect(component).toMatchSnapshot('children display');
  });
});
