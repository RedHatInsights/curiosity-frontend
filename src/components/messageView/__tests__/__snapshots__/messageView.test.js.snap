// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageView Component should have fallback conditions for all props: fallback display 1`] = `
<PageLayout>
  <PageHeader>
    Subscription Watch
  </PageHeader>
  <EmptyState
    className="fadein"
    variant="full"
  />
</PageLayout>
`;

exports[`MessageView Component should render a non-connected component: non-connected 1`] = `
<PageLayout>
  <PageHeader>
    Subscription Watch
  </PageHeader>
  <EmptyState
    className="fadein"
    variant="full"
  >
    <EmptyStateIcon
      icon={[Function]}
    />
    <Title
      headingLevel="h2"
      size="lg"
    >
      lorem
    </Title>
    <EmptyStateBody>
      ipsum
    </EmptyStateBody>
  </EmptyState>
</PageLayout>
`;
